File: src/main/java/application/history/history/application/services/GetHistoryByDateService.java
package application.history.history.application.services;

import application.history.history.domain.entities.History;
import application.history.history.domain.repositories.HistoryRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.util.Date;
import java.util.Optional;

@Service
public class GetHistoryByDateService {
    private final HistoryRepository repository;

    public GetHistoryByDateService(HistoryRepository repository) {
        this.repository = repository;
    }

    public Optional<History> run(LocalDate date) {
        return this.repository.findByDate(date);
    }
}

File: src/history/main/kotlin/com/example/history/application/services/GetHistoryByDateService.kt
package com.example.history.application.services

import com.example.history.domain.entities.History
import com.example.history.domain.repositories.HistoryRepository
import org.springframework.stereotype.Service
import java.time.LocalDate
import java.util.*

@Service
class GetHistoryByDateService(repository: HistoryRepository) {
    private val repository: HistoryRepository

    fun run(date: LocalDate): Optional<History> {
        return repository.findByDate(date)
    }

    init {
        this.repository = repository
    }
}
